ui/components/app/permissions-connect-permission-list/permissions-connect-permission-list.js

```diff
@@ -1,37 +1,40 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { getWeightedPermissions } from '../../../helpers/utils/permission';
+import {
+  getRightIcon,
+  getWeightedPermissions,
+} from '../../../helpers/utils/permission';
 import { useI18nContext } from '../../../hooks/useI18nContext';
-import PermissionCell from '../permission-cell';
-import Box from '../../ui/box';
 
-export default function PermissionsConnectPermissionList({
-  permissions,
-  targetSubjectMetadata,
-}) {
+/**
+ * Get one or more permission descriptions for a permission name.
+ *
+ * @param permission - The permission to render.
+ * @param index - The index of the permission.
+ * @returns {JSX.Element} A permission description node.
+ */
+function getDescriptionNode(permission, index) {
+  const { label, leftIcon, permissionName } = permission;
+
+  return (
+    <div className="permission" key={`${permissionName}-${index}`}>
+      {typeof leftIcon === 'string' ? <i className={leftIcon} /> : leftIcon}
+      {label}
+      {getRightIcon(permission)}
+    </div>
+  );
+}
+
+export default function PermissionsConnectPermissionList({ permissions }) {
   const t = useI18nContext();
 
   return (
-    <Box paddingTop={2} paddingBottom={2}>
-      {getWeightedPermissions(t, permissions, targetSubjectMetadata).map(
-        (permission, index) => {
-          return (
-            <PermissionCell
-              title={permission.label}
-              description={permission.description}
-              weight={permission.weight}
-              avatarIcon={permission.leftIcon}
-              dateApproved={permission?.permissionValue?.date}
-              key={`${permission.permissionName}-${index}`}
-            />
-          );
-        },
-      )}
-    </Box>
+    <div className="permissions-connect-permission-list">
+      {getWeightedPermissions(t, permissions).map(getDescriptionNode)}
+    </div>
   );
 }
 
 PermissionsConnectPermissionList.propTypes = {
   permissions: PropTypes.object.isRequired,
-  targetSubjectMetadata: PropTypes.object.isRequired,
```

Comment:
Keep `targetSubjectMetadata` and pass it to `getWeightedPermissions`. Otherwise we may have weird errors in the future.

Reply:
I think that this is redundant now, because we're not using this component for snap permissions anymore.
The only usage of it now, is here: https://github.com/MetaMask/metamask-extension/blob/c67bb6b03496889220d45c9bafdc41b144992cfc/ui/components/app/permission-page-container/permission-page-container-content/permission-page-container-content.component.js#L35-L35

The snap permission list component is keeping this subject metadata as param.
I can return it here also, but I'm not sure what metadata I should pass here in this case if you see a need for it?

Reply:
Any permission description may use it in the future since it is required in `getWeightedPermissions`: https://github.com/MetaMask/metamask-extension/blob/c02908a77eb123471f6e50c6766816b39f114480/ui/helpers/utils/permission.js#L481

Therefore I think we should always pass it, even if not used currently by non snap permissions. That way we prevent future bugs.

Reply:
Alright, added it. Let me know if anything around it needs to be different.